card/lib/card/view/cache.rb

```diff
@@ -31,58 +31,73 @@ class View
+    # anything about the card itself. So when I update the card "Johnny", all the cached
+    # views of "Johnny" are cleared. Similarly, changes to structure rules and other
     # basic patterns are typically well managed by the caching system.
     #
-    # However, a card is generally far less smart about clearing its own cache when
-    # changes happen to other cards that affect a rule via _logic internal to the view_.
+    # However, there are many other potential changes that views cannot detect. Views that
+    # are susceptible to these "cache hazards" should be configured with `cache: :never`.
     #
-    # For example, consider the following view:
+    # ## Cache hazards
     #
-    #     view :myview do
-    #       Card["random"].content
-    #     end
+    # If a view contains any of the following cache hazards, it would be wise to consider
+    # a `cache: :never` configuration:
     #
-    # If this view is cached, and then the card "random" changes, the caching system does
-    # not know to clear this cache, so it may be wise to set `cache: :never` in the
-    # view definition.
+    # - a dynamic search (eg `Card.search`) whose results may change
+    # - a live timestamp (eg `Time.now`)
+    # - environmental variables (eg `Env.params`)
+    # - any variables altered in one view and used in another (eg `@myvar`)
+    # - other cards' properties (eg `Card["random"].content`)
     #
-    # Some other common situations likely to require `cache: :never`:
+    # What all of the above have in common is that they involve changes about which the
+    # view caching system is unaware. This means that whether the cache hazard is
+    # rendered directly in a view or just used in its logic, it can change in a way
+    # that _should_ change the view but _won't_ change the view if it's cached.
     #
-    # 1. view manipulates another rendered view. If the other view generates a stub
-    #    then the manipulating view will find itself manipulating a stub.
+    # ## Altering cached views
+    #
+    # Whereas ignoring cache hazards may cause over-caching, altering cached views
+    # may cause outright errors. If a view directly alters a rendered view,
+    # it may be dangerous to cache.
     #
     #        # obviously safe to cache
     #        view(:x) { "ABC" }
     #
-    #        # also safe, because x is not manipulated
+    #        # also safe, because x is NOT altered
     #        view(:y) { render_x + "DEF" }
     #
-    #        # unsafe, because x is manipulated
+    #        # unsafe and thus never cached, because x is altered
     #        view(:z, cache: :never) { render_z.reverse }
     #
-    # 2. view performs a dynamic search (eg via Card.search) whose results may change
-    #
-    # 3. view displays a timestamp
+    # Specifically, the danger is that the inner view will be rendered as a stub,
+    # and the out view will end up altering the stub and not the view.
```

Comment:
ok, that explains it. Kind of. ~~Why does `render_x` return a stub? I thought we stub unsafe stuff like nests. Isn't another view of the same card safe and we get the cached view?~~

A thought: If `render_x` can return a stub is it maybe better to have api to "force render" (never give me a stub) instead of marking the parent view as unsafe?

Reply:
if there's some reason to render x as a stub, that means we think it's not safe to cache it in another view. So a forcing a render and then caching it is more dangerous than not caching.